src/main/resources/shaders/oxygen.glsl.vert

```diff
@@ -0,0 +1,58 @@
+#version 330 core
+
+layout (location = 0) in vec3  aPosition;
+layout (location = 1) in vec3  aNormal;
+layout (location = 2) in vec3  aInstancePosition;
+layout (location = 3) in float aInstanceTemperature;
+layout (location = 4) in int   aInstanceMaterial;
+layout (location = 5) in int   aInstanceBurningTime;
+layout (location = 6) in float aInstanceSmokeIndicator;
+layout (location = 7) in float aInstanceOxygenLevel;
+
+uniform mat4 uProjection;
+uniform mat4 uView;
+
+out vec3 fNormal;
+out vec4 fColor;
+
+const vec4 DEBUG_COLOR = vec4(1.0, 0.0, 1.0, 0.5);
+const vec4 TRANSPARENT_COLOR = vec4(0.0);
+const vec4 THIN_COLOR = vec4(0.7, 0.7, 0.7, 0.01);
+const vec4 DENSE_COLOR  = vec4(0.0, 0.0, 1.0, 0.3);
```

Comment:
Try using other colors, when I run the simulation I can barely see the difference